src/main/webapp/resources/js/pages/UserGroupsPage/components/CreateNewUserGroupButton.jsx

```diff
@@ -1,22 +1,32 @@
-import React, { useContext, useEffect, useRef, useState } from "react";
+import React, { useEffect, useRef, useState } from "react";
 import { AddNewButton } from "../../../components/Buttons/AddNewButton";
 import { Form, Input, Modal } from "antd";
 import { createUserGroup } from "../../../apis/users/groups";
 import { useNavigate } from "@reach/router";
+import { useResetFormOnCloseModal } from "../../../hooks";
 
 /**
  * React component to render a button to create a new user group
  * @returns {*}
  * @constructor
  */
-export function CreateNewUserGroupButton() {
+export function CreateNewUserGroupButton({ baseUrl }) {
   /*
   Required a reference to the user select input so that focus can be set
   to it when the window opens.
    */
   const inputRef = useRef();
   const [visible, setVisible] = useState();
+
+  /*
+  Ant Design form
+   */
   const [form] = Form.useForm();
+  useResetFormOnCloseModal({
+    form,
+    visible,
+  });
+
   const navigate = useNavigate();
 
   const [error, setError] = useState();
```

Comment:
Can you move this to the top of the function, I like to have all the useState's together